### ISSUE TITLE
loaded model gets different loss than saved one

### ISSUE THREAD

**Issue by yftah89** [keras-team/keras#9933]
I'm using the callback ModelCheckpoint with save_best_only=True.
When I load the model right after training the evaluating seems to be different than model.fit 
val_score, val_acc = model.evaluate(val_data, Y_val, batch_size=100)
print('val loss:', val_score)

the val_score which is the val loss is different than the best val lost that keras is printing
